### PR TITLE
Gets React/Apollo working

### PR DESCRIPTION
**Pull request by dzucconi**

### CODE REVIEW

File: apps/graphql/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const app = module.exports = express();
+
+app.post('/graphql', (req, res, next) => {
+  if (NODE_ENV !== 'development' && req.hostname !== 'are.na') {
+    const err = new Error('Requests must originate from are.na');
+    return next(err);
+  }
+
+  const X_AUTH_TOKEN = req.user && req.user.get('authentication_token') || '';
```
Comment (dzucconi) [inline apps/graphql/index.js]: This only works if we're on the server.

File: lib/apollo/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { ApolloClient } from 'apollo-client';
+import { createHttpLink } from 'apollo-link-http';
+import { setContext } from 'apollo-link-context';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+
+const httpLink = createHttpLink({ uri: '/graphql' });
+
+const authLink = setContext((_, { headers }) => {
+  const { data: { CURRENT_USER } } = sharify;
+  const X_AUTH_TOKEN = CURRENT_USER && CURRENT_USER.authentication_token || '';
```
Comment (dzucconi) [inline lib/apollo/index.js]: And this only works if we're on the client.

**Comment by dzucconi** [comment]
react

File: apps/graphql/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+import express from 'express';
+
+import { NODE_ENV, GRAPHQL_ENDPOINT, X_APP_TOKEN } from 'config.coffee';
+
+const app = module.exports = express();
+
+app.post('/graphql', (req, res, next) => {
+  if (NODE_ENV !== 'development' && req.hostname !== 'are.na') {
+    const err = new Error('Requests must originate from are.na');
```
Comment (broskoski) [inline apps/graphql/index.js]: 👍

**Pull request merged**